test/library/keypress_and_caret.spec.js

```diff
@@ -342,6 +342,24 @@ describe('Test keypress and caret position changes', () => {
       jasmine.clock().uninstall()
     });
 
+    it('should clear active timers', () => {
+      jasmine.clock().install();
+
+      const wrapper = mount(<NumberFormat />);
+      const instance = wrapper.instance();
+
+      simulateFocusEvent(wrapper.find('input'), 0, 0, setSelectionRange);
+
+      expect(instance.focusTimeout).toBeTruthy();
+
+      wrapper.unmount();
+      jasmine.clock().tick(1);
+
+      expect(instance.focusTimeout).toBeFalsy();
```

Comment:
Instead of checking if focusTimeout is empty let's check if the function inside the timeout is called. We can use onFocus prop to check this.